### DISCUSSION TITLE
[Logs] How do I get a simplified logger result?

### DISCUSSION THREAD

**Discussion by MateuszMiekicki** [open-telemetry/opentelemetry-cpp#1747] [Q&A] [answered]
Hi. I managed to read the examples, I also found a LogHandler, but it doesn't match the result of the LoggerProvider. Generallyi I would like to keep the fields from the "complex" logger, but at the same time I need the file name and the line. Can I get this somehow?

```cpp
#include "opentelemetry/exporters/ostream/log_record_exporter.h"
#include "opentelemetry/logs/provider.h"
#include "opentelemetry/sdk/logs/logger_provider.h"
#include "opentelemetry/sdk/logs/simple_log_record_processor.h"
#include <memory>
namespace logs  = opentelemetry::logs;
namespace sdklogs      = opentelemetry::sdk::logs;
namespace logs_api     = opentelemetry::logs;
namespace exporterlogs = opentelemetry::exporter::logs;

int main()
{
  auto exporter = std::unique_ptr<sdklogs::LogRecordExporter>(new exporterlogs::OStreamLogExporter(std::cout));
  auto sdkProvider = std::shared_ptr<sdklogs::LoggerProvider>(new sdklogs::LoggerProvider());
  sdkProvider->AddProcessor(std::unique_ptr<sdklogs::LogRecordProcessor>(
      new sdklogs::SimpleLogProcessor(std::move(exporter))));
  auto apiProvider = std::shared_ptr<logs_api::LoggerProvider>(sdkProvider);
  auto provider    = std::shared_ptr<logs_api::LoggerProvider>(apiProvider);
    auto logger = provider->GetLogger("firstlog", "", "my lib", "12.32.2");
  logger->Debug("I am the first log message.");
}
```
I have something like this. It is also possible that I don't understand the premise of this logger, then I would ask for a reference to the documentation. Thanks

**Comment by lalitb** [comment, marked as answer]
> I would like to keep the fields from the "complex" logger, but at the same time I need the file name and the line.

These fields are coming from the logs [data model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#log-and-event-record-definition). If you want to add file-name/line-number, you can pass them using attributes field. We may plan to create convenience macros on top of the existing API to implicitly add these fields, but at the moment Logging API/SDK is still experimental, and work in progress - https://github.com/open-telemetry/opentelemetry-cpp/milestone/18, and not recommended for production use.

> I also found a LogHandler, but it doesn't match the result of the LoggerProvider.

These are two different things - LogHandler is part of opentelemetry-cpp internal logging, while LoggerProvider is what you   need to use for instrumenting your application.

**Reply by MateuszMiekicki** [reply]
I know it's a bit of a fortune teller, but is this 11 November realistic for a logger release? I would like to use the whole opentelemtry 'package' in an upcoming project. Hence my question.

**Reply by lalitb** [reply]
Yeah, it's definitely not realistic date, don't see the RC be ready before this month. Also the OpenTelemetry Logging specs is still [experimental](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/logs), so even after RC, breaking changes are expected if specs changes.

**Reply by MateuszMiekicki** [reply]
Thanks, this will make my choice of tools easier.
